src/sink/blocking.rs

```diff
@@ -0,0 +1,114 @@
+use {Async, AsyncSink};
+use sink::Sink;
+use executor;
+use task_impl::ThreadNotify;
+
+/// Provides thread-blocking operations on a sink.
+///
+/// See [`blocking`](fn.blocking.html) documentation for more details.
+#[derive(Debug)]
+#[must_use = "sinks do nothing unless used"]
+pub struct Blocking<T: Sink> {
+    inner: Option<executor::Spawn<T>>,
+}
+
+/// Provides thread-blocking operations on a sink.
+///
+/// `blocking` consumes ownership of `sink`, returning a `Blocking` backed by
+/// the sink. The `Blocking` value exposes thread-blocking operations that allow
+/// sending values into the sink. For example, `Blocking::send` will block the
+/// current thread until the inner sink has capacity to accept the item being
+/// sent.
+///
+/// **These operations will block the current thread**. This means that they
+/// should not be called while in the context of a task executor as this will
+/// block the task executor's progress. Also, note that **this value will block
+/// the current thread on drop**. Droping `Blocking` will ensure that the inner
```

Comment:
Might be worth mentioning that this is analogous to dropping a `BufWriter` in `std`.